### PR TITLE
[Storage] Remove breaking change warning message when creating a storage account

### PR DESCRIPTION
**Pull request by yifanz7**
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Storage/Storage.Management/StorageAccount/NewAzureStorageAccount.cs [hunk 1]
```diff
@@ -481,7 +475,7 @@ public int KeyExpirationPeriodInDay
 
         [Parameter(
             Mandatory = false,
-            HelpMessage = "Allow public access to all blobs or containers in the storage account. The default interpretation is true for this property.")]
+            HelpMessage = "Allow public access to all blobs or containers in the storage account. The default interpretation is false for this property.")]
```
Comment (blueww) [inline src/Storage/Storage.Management/StorageAccount/NewAzureStorageAccount.cs:478]: Pleas also update the help file.

File: src/Storage/Storage.Management/ChangeLog.md [hunk 2]
```diff
@@ -19,6 +19,8 @@
 -->
 ## Upcoming Release
 * Updated Azure.Core to 1.34.0.
+* Removed the breaking change warning message for creating a storage account. The default interpretation of AllowBlobPublicAccess and AllowCrossTenantReplication has been changed to false
```
Comment (blueww) [inline src/Storage/Storage.Management/ChangeLog.md:22]: As we agreed , the change log should link to https://techcommunity.microsoft.com/t5/azure-storage-blog/azure-storage-updating-some-default-security-settings-on-new/ba-p/3819554

I would suggest also send the PR link to the feature PM by mail to review . Mail title "CLI/Powershell change for disable anonymous access/cross-tenant replication"

**Comment by blueww** [comment]
SRP rollout for anonymous access is paused with no new ETA.
Cross-tenant is also paused.

So we should not merge this PR currently

**Comment by yifanz7** [comment]
> SRP rollout for anonymous access is paused with no new ETA. Cross-tenant is also paused.
> 
> So we should not merge this PR currently

Will close the PR for now

**Pull request closed without merging**